### ISSUE TITLE
_ in URL makes them invalid

### ISSUE THREAD

**Issue by Gouthamve** [asaskevich/govalidator#212]
See https://github.com/prometheus/prometheus/issues/2717

From that issue:
[RFC 952](https://tools.ietf.org/html/rfc952) says that:
```
1. A "name" (Net, Host, Gateway, or Domain name) is a text string up
   to 24 characters drawn from the alphabet (A-Z), digits (0-9), minus
   sign (-), and period (.)
```
> But the rule apparently isn't strictly followed by DNS servers and names in an /etc/hosts file can contain underscores.

Do you have any plans to support `_`?

**Comment by brian-brazil** [comment]
Did #194 not cover this?

**Comment by Gouthamve** [comment]
Hmm, interestingly:
`govalidator.IsURL("aio1_alertmanager_container-63376c45:9093")` ---> `false`

And I did do a `go get -u`

@brian-brazil Can you try this?
```
package main

import (
	"fmt"

	"github.com/asaskevich/govalidator"
)

func main() {
	fmt.Println(govalidator.IsURL("aio1_alertmanager-container-63376c45:9093"))
}
```

**Comment by brian-brazil** [comment]
With the latest code, that's a false.

**Comment by pgier** [comment]
It's false because of the call to url.Parse (https://github.com/asaskevich/govalidator/blob/master/validator.go#L57).  This will return an error for something like "my_server.example.com:80", but will accept both "http://my_server.example.com:80" and "myserver.example.com:80".  The first colon in the string is interpreted as the scheme delimiter, unless it contains an invalid character such as an underscore, in which case it tries to interpret the string as a relative path which cannot contain a colon.

So you hit a combination of including a colon and underscore, without including the "//" to indicate a hostname.
Some examples: https://play.golang.org/p/VLHe7vPO1I

In go 1.7 and earlier this didn't cause an error, but it changed to be more strict in 1.8 (https://github.com/golang/go/commit/c5ccbdd22bdbdc43d541b7e7d4ed66ceb559030e)

**Comment by jasonlam604** [comment]
Submitted pull request #240

**Comment by asaskevich** [comment]
Hope it is already fixed in #240, isn't it?

**Comment by asaskevich** [comment]
@Gouthamve https://github.com/asaskevich/govalidator/commit/6c3cb3214c0bf1206da40f9cb4eef372a5788307#diff-bf215ece4e5166da723b1e19a0d6b1cdR667 resolved, sure
